Install script fails to find "lvim" command

### Problem description

Running installer (in zsh) will fail on the "verifying core plugins" step:
`Verifying core plugins
/home/xxx/.local/share/lunarvim/lvim/utils/ci/verify_plugins.sh: line 9: lvim: command not found`

It fails because lvim is not found since ~/.local/bin has not been added to PATH.

In my case, solution is to manually add ~/.local/bin to .zshrc PATH and re-run install.  Not sure if ensuring that this path is included should be done by installer or is expected to be actionable by the user.

Running Arch & zsh w/ohmyzsh.


### LunarVim version

happens on latest stable and rolling

### Neovim version (>= 0.7.2)

0.7.2

### Operating system/version

Arch Linux

### Steps to reproduce

1. simply run installer

### support info

Verifying core plugins
/home/xxx/.local/share/lunarvim/lvim/utils/ci/verify_plugins.sh: line 9: lvim: command not found

### Screenshots

_No response_

Hello, I've tried it on v7.0.0 in docker, and it doesn't work for me either:
```
FROM ubuntu

RUN apt-get update

RUN apt-get install -y curl git make

RUN curl -L https://github.com/neovim/neovim/releases/download/v0.7.0/nvim-linux64.deb >/nvim.deb \
 && apt install /nvim.deb

RUN curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh | bash -

ENTRYPOINT nvim
```

```
❯ docker build -t lunarvim .
[+] Building 12.2s (8/8) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                      0.0s
 => => transferring dockerfile: 498B                                                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                                                                           0.0s
 => [internal] load metadata for docker.io/library/ubuntu:latest                                                                                                                                          0.0s 
 => [1/5] FROM docker.io/library/ubuntu                                                                                                                                                                   0.0s
 => CACHED [2/5] RUN apt-get update                                                                                                                                                                       0.0s 
 => CACHED [3/5] RUN apt-get install -y curl git make                                                                                                                                                     0.0s 
 => [4/5] RUN curl -L https://github.com/neovim/neovim/releases/download/v0.7.0/nvim-linux64.deb >/nvim.deb  && apt install /nvim.deb                                                                     4.0s 
 => ERROR [5/5] RUN curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh | bash -                                                                                8.0s
------
 > [5/5] RUN curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh | bash -:
#8 0.496
#8 0.496       88\                                                   88\
#8 0.496       88 |                                                  \__|
#8 0.496       88 |88\   88\ 888888$\   888888\   888888\ 88\    88\ 88\ 888888\8888\
#8 0.496       88 |88 |  88 |88  __88\  \____88\ 88  __88\\88\  88  |88 |88  _88  _88\
#8 0.496       88 |88 |  88 |88 |  88 | 888888$ |88 |  \__|\88\88  / 88 |88 / 88 / 88 |
#8 0.496       88 |88 |  88 |88 |  88 |88  __88 |88 |       \88$  /  88 |88 | 88 | 88 |
#8 0.496       88 |\888888  |88 |  88 |\888888$ |88 |        \$  /   88 |88 | 88 | 88 |
#8 0.496       \__| \______/ \__|  \__| \_______|\__|         \_/    \__|\__| \__| \__|
#8 0.496
#8 0.498 --------------------------------------------------------------------------------
#8 0.498 Detecting platform for managing any additional neovim dependencies
#8 0.510 --------------------------------------------------------------------------------
#8 0.510 Would you like to install LunarVim's NodeJS dependencies?
#8 0.512 --------------------------------------------------------------------------------
#8 0.512 Would you like to install LunarVim's Python dependencies?
#8 0.514 --------------------------------------------------------------------------------
#8 0.514 Would you like to install LunarVim's Rust dependencies?
#8 0.520 --------------------------------------------------------------------------------
#8 0.520 Cloning LunarVim configuration
#8 0.522 Cloning into '/root/.local/share/lunarvim/lvim'...
#8 1.393 --------------------------------------------------------------------------------
#8 1.394 Installing LunarVim shim
#8 1.396 make: Entering directory '/root/.local/share/lunarvim/lvim'
#8 1.399 starting LunarVim bin-installer
#8 1.400 bash ./utils/installer/install_bin.sh
#8 1.415 You can start LunarVim by running: /root/.local/bin/lvim
#8 1.416 make: Leaving directory '/root/.local/share/lunarvim/lvim'
#8 1.421 Preparing Packer setup
#8 2.330 [packer.nvim] [DEBUG 09:57:51] packer.lua:531: packer.sync: requiring modules[packer.nvim] [DEBUG 09:57:51] packer.lua:327: Processing plugin specs[packer.nvim] [DEBUG 09:57:51] update.lua:56: Fixing plugin types[packer.nvim] [DEBUG 09:57:51] update.lua:67: Done fixing plugin types[packer.nvim] [DEBUG 09:57:51] clean.lua:22: Starting clean[packer.nvim] [INFO  09:57:51] clean.lua:82: Already clean![packer.nvim] [DEBUG 09:57:51] packer.lua:561: Gathering install tasks[packer.nvim] [DEBUG 09:57:51] packer.lua:564: Gathering update tasks[packer.nvim] [DEBUG 09:57:51] packer.lua:568: Gathering luarocks tasks[packer.nvim] [DEBUG 09:57:51] packer.lua:581: Running tasks[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed hrsh7th/cmp-nvim-lsp[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed nvim-lua/popup.nvim[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed saadparwaiz1/cmp_luasnip[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed ahmedkhalf/project.nvim[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed rcarriga/nvim-notify[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed max397574/which-key.nvim[packer.nvim] [DEBUG 09:57:53] install.lua:23: Installed antoinemadec/FixCursorHold.nvim[packer.nvim] [DEBUG 09:57:53] install.lua:26: Failed to install nvim-telescope/telescope-fzf-native.nvim: {[packer.nvim]   data = {[packer.nvim]     exit_code = 2,[packer.nvim]     signal = 0[packer.nvim]   },[packer.nvim]   msg = "Error running post update hook: mkdir -p build^@cc -O3 -Wall -Werror -fpic -std=gnu99 -shared src/fzf.c -o build/libfzf.so^@make: cc: No such file or directory^@make: *** [Makefile:18: build/libfzf.so] Error 127"[packer.nvim] }[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed hrsh7th/cmp-buffer[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed lunarvim/onedarker.nvim/freeze[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed kyazdani42/nvim-web-devicons[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed Tastyep/structlog.nvim[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed hrsh7th/nvim-cmp[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed numToStr/Comment.nvim[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed hrsh7th/cmp-path[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed goolord/alpha-nvim[packer.nvim] [DEBUG 09:57:54] install.lua:23: Installed b0o/schemastore.nvim[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed tamago324/nlsp-settings.nvim[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed windwp/nvim-autopairs[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed max397574/lua-dev.nvim[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed JoosepAlviste/nvim-ts-context-commentstring[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed nvim-lualine/lualine.nvim[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed williamboman/nvim-lsp-installer[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed lewis6991/gitsigns.nvim[packer.nvim] [DEBUG 09:57:55] install.lua:23: Installed akinsho/toggleterm.nvim/main[packer.nvim] [DEBUG 09:57:56] update.lua:88: Updated wbthomason/packer.nvim: {[packer.nvim]   err = {},
#8 7.219 Error executing vim.schedule lua callback: ...im/site/pack/packer/start/packer.nvim/lua/packer/log.lua:109: Vim(echomsg):E121: Undefined variable: compile
#8 7.223 stack traceback:
#8 7.223        [C]: in function 'cmd'
#8 7.223        ...im/site/pack/packer/start/packer.nvim/lua/packer/log.lua:109: in function ''
#8 7.223        vim/_editor.lua: in function <vim/_editor.lua:0>[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed neovim/nvim-lspconfig[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed jose-elias-alvarez/null-ls.nvim[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed nvim-telescope/telescope.nvim/0.1.x[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed akinsho/bufferline.nvim/main[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed nvim-treesitter/nvim-treesitter[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed rafamadriz/friendly-snippets[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed L3MON4D3/LuaSnip[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed nvim-lua/plenary.nvim[packer.nvim] [DEBUG 09:57:56] install.lua:23: Installed kyazdani42/nvim-tree.lua[packer.nvim] [DEBUG 09:57:56] packer.lua:327: Processing plugin specs[packer.nvim] [INFO  09:57:56] packer.lua:739: Finished compiling lazy-loaders![packer.nvim] [DEBUG 09:57:56] packer.lua:347: packer.compile: CompletePacker setup complete
#8 7.916 --------------------------------------------------------------------------------
#8 7.916 Verifying core plugins
#8 7.923 /root/.local/share/lunarvim/lvim/utils/ci/verify_plugins.sh: line 9: lvim: command not found
#8 7.923 [ERROR]: Unable to verify plugins, make sure to manually run ':PackerSync' when starting lvim for the first time.
------
executor failed running [/bin/sh -c curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh | bash -]: exit code: 1
```

--- a/utils/installer/install.sh
+++ b/utils/installer/install.sh
@@ -299,7 +299,6 @@ function __validate_node_installation() {
   fi
 
   if [ ! -d "$manager_home" ] || [ ! -w "$manager_home" ]; then
-    echo "[ERROR] Unable to install using [$pkg_manager] without administrative privileges."
     return 1
   fi
 
@@ -314,21 +313,21 @@ function install_nodejs_deps() {
       return
     fi
   done
-  print_missing_dep_msg "${pkg_managers[@]}"
-  exit 1
+  echo "[WARN]: skipping installing optional nodejs dependencies due to insufficient permissions."
+  echo "check how to solve it: https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally"
 }
 
 function install_python_deps() {
   echo "Verifying that pip is available.."
-  if ! python3 -m ensurepip &>/dev/null; then
+  if ! python3 -m ensurepip >/dev/null; then
     if ! python3 -m pip --version &>/dev/null; then
-      print_missing_dep_msg "pip"
-      exit 1
+      echo "[WARN]: skipping installing optional python dependencies"
+      return 1
     fi
   fi
   echo "Installing with pip.."
   for dep in "${__pip_deps[@]}"; do
-    python3 -m pip install --user "$dep"
+    python3 -m pip install --user "$dep" || return 1
   done
   echo "All Python dependencies are successfully installed"
 }

--- a/utils/installer/install.sh
+++ b/utils/installer/install.sh
@@ -26,6 +26,7 @@ declare ARGS_LOCAL=0
 declare ARGS_OVERWRITE=0
 declare ARGS_INSTALL_DEPENDENCIES=1
 declare INTERACTIVE_MODE=1
+declare ADDITIONAL_WARNINGS=""
 
 declare -a __lvim_dirs=(
   "$LUNARVIM_CONFIG_DIR"
@@ -148,6 +149,7 @@ function main() {
 
   setup_lvim
 
+  msg "$ADDITIONAL_WARNINGS"
   msg "Thank you for installing LunarVim!!"
   echo "You can start it by running: $INSTALL_PREFIX/bin/lvim"
   echo "Do not forget to use a font with glyphs (icons) support [https://github.com/ryanoasis/nerd-fonts]"
@@ -226,7 +228,25 @@ function validate_lunarvim_files() {
   fi
 }
 
+function validate_install_prefix() {
+  local prefix="$1"
+  case $PATH in
+    *"$prefix/bin"*)
+      return
+      ;;
+  esac
+  local profile="$HOME/.profile"
+  test -z "$ZSH_VERSION" && profile="$HOME/.zshenv"
+  ADDITIONAL_WARNINGS="[WARN] the folder $prefix/bin is not on PATH, consider adding 'export PATH=$prefix/bin:\$PATH' to your $profile"
+
+  # avoid problems when calling any verify_* function
+  export PATH="$prefix/bin:$PATH"
+}
+
 function check_system_deps() {
+
+  validate_install_prefix "$INSTALL_PREFIX"
+
   if ! command -v git &>/dev/null; then
     print_missing_dep_msg "git"
     exit 1

